### ISSUE TITLE
Fix assertion failure upon running

### ISSUE THREAD

**Issue by asialasr** [asialasr/OpenGL2dEx#2]
There's an assertion failure that happens seemingly randomly upon starting the program that needs to be fixed

**Comment by asialasr** [comment]
I've found a solution for this that I'm moving forward with; it's hard to "prove" it's the solution since it is slightly random (although the error occurs more often than not).  I've run quite a few times with no issue thus far.

The ERROR 1280 (GL_INVALID_ENUM) in glGetIntegerV shows up just after gladLoadGLLoader is called.  The error happens in gladLoadGLLoader > find_extensionsGL > get_exts() > glGetIntegerV @ line 210.  Originally, I thought the GL context may somehow not be major version 3+ because GL_NUM_EXTENSIONS was added as a valid enum for glGetIntegerV in major version 3, I believe.  However, it looked like the correct/expected version was loading, so I found that using the compatibility profile instead of the core profile fixed the issue I was seeing.  My gut reaction based on some minor research is that this was deprecated at some point, which would explain why compatibility works but core does not.  However, this is a very preliminary thought and requires more investigation.
